src/block/block_ckpt.c

```diff
@@ -1084,6 +1084,57 @@ __wt_block_checkpoint_resolve(WT_SESSION_IMPL *session, WT_BLOCK *block, bool fa
     return (ret);
 }
 
+/*
+ * __wti_block_checkpoint_extlist_dump --
+ *     Dump all of the available checkpoints extent lists, excluding the specified offset. Extent
+ *     offsets should never be 0, as that is the offset of the file header.
+ */
+int
+__wti_block_checkpoint_extlist_dump(WT_SESSION_IMPL *session, WT_BLOCK *block)
+{
+    WT_BLOCK_CKPT *ci;
+    WT_CKPT *ckpt_iter, *ckptbase;
+    WT_DECL_RET;
+    size_t ckpt_bytes_allocated;
+
+    ckptbase = NULL;
+
+    WT_ERR(__wt_meta_ckptlist_get(
+      session, session->dhandle->name, false, &ckptbase, &ckpt_bytes_allocated));
+    WT_CKPT_FOREACH (ckptbase, ckpt_iter) {
+        WT_ERR(__wt_calloc(session, 1, sizeof(WT_BLOCK_CKPT), &ckpt_iter->bpriv));
+        ci = ckpt_iter->bpriv;
+
+        WT_ERR(__wti_block_ckpt_init(session, ci, ckpt_iter->name));
+        WT_ERR(
+          __wti_block_ckpt_unpack(session, block, ckpt_iter->raw.data, ckpt_iter->raw.size, ci));
+
+        if (ci->alloc.offset != WT_BLOCK_INVALID_OFFSET &&
+          __wti_block_extlist_read(session, block, &ci->alloc, ci->file_size) == 0)
+            WT_TRET(__wti_block_extlist_dump(session, block, &ci->alloc));
+
+        if (ci->avail.offset != WT_BLOCK_INVALID_OFFSET &&
+          __wti_block_extlist_read(session, block, &ci->avail, ci->file_size) == 0)
+            WT_TRET(__wti_block_extlist_dump(session, block, &ci->avail));
+
+        if (ci->discard.offset != WT_BLOCK_INVALID_OFFSET &&
+          __wti_block_extlist_read(session, block, &ci->discard, ci->file_size) == 0)
+            WT_TRET(__wti_block_extlist_dump(session, block, &ci->discard));
+
+        WT_ERR(ret);
```

Comment:
@Sean04, if we encounter an error, we may exit here which means we may miss on the other checkpoints?

Reply:
yes, I think that's okay. My thinking here is that if we fail initialising the checkpoint struct we probably won't be able to initialise the next one anyway.